Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+        let requestParam = request.params.datasetId;
+        let datasetId = Number(requestParam);
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            let userId: number = request.body.user.account_id
+            try {
+                let requestResponse = await this.dataSetService.removeSavedDatasetService(userId, datasetId)
+                return response.status(requestResponse.statusCode).send(requestResponse.message);
+            } catch (err) {
+                response.status(err.status).send(err.message);
+            }
+        }
+    }
+
+    /**
+     * This controller will take a request, grab the user ID, and if it a real number will then process the request,
+     * send it to the getDataService to acquire an array of data sets that were uploaded by this user ID, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUserUploadedDatasets(request: Request, response: Response) {
+        let userId: any = request.body.user.account_id
+        if (isNaN(userId)) {
+            response.status(400).send("Invalid search params entered");
+        }
+        else {
+            try {
+                let arrayOfData = await this.dataSetService.getUserUploadedDatasets(userId)
+                return response.status(200).json(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    /**
+     * This controller will take a request, grab the user ID, and if it a real number will then process the request,
+     * send it to the getDataService to acquire an array of data sets that were saved by this user ID, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUserSavedDatsets(request: Request, response: Response) {
+        let userId: any = request.body.user.account_id
+        if (isNaN(userId)) {
+            response.status(500).json("Invalid search params entered");
+        }
+        else {
+            try {
+                let arrayOfData = await this.dataSetService.getUserSavedDatasets(userId)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    //TODO: Readd later

```

Comment:
Readd later todo?

Reply:
It was accidentally removed at some point, was thus not updated as changes to this controller were made, and as the frontend plans for these features was dropped, I added this back in as a comment block for next sprint.